### PR TITLE
Send email when Supervisor assigned / removed

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

**Comment by zenith19** [comment]
Ok

File: app/controllers/supervisor/users_courses_controller.rb [hunk 1]
```diff
@@ -6,7 +6,13 @@ def edit
   end
 
   def update
+    old_user_ids = load_old_user_ids
     if @course.update_attributes course_params
+      users = calculate_changes old_user_ids
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_courses_controller.rb]: use call back funtion
after_create and after_destroy in UserCourse model

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    @course = Course.find course_id
+    @user_full_name = user_full_name
+    mail to: user_email, subject: default_i18n_subject(course_name: @course.name)
+  end
+
+  def removed_supervisors users, course
+    @course = course    
+    users.each do |user|
+      @user = user
+      mail to: @user.email, subject: "This is a test email for removed users"
```
Comment (chuanhtuan) [inline app/mailers/user_mailer.rb]: i18n

File: app/mailers/application_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "noreply.esv@gmail.com"
+  layout 'mailer'
```
Comment (MuhammadTamzid) [inline app/mailers/application_mailer.rb]: Double quote
layout "mailer"

File: app/views/user_mailer/assigned_supervisors.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/assigned_supervisors.html.erb]: Double quote

File: app/views/user_mailer/assigned_supervisors.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
+    <h1>Welcome to <%= I18n.t :system_name %></h1>    
+    <p>
+      You have been assigned to a course.
+    </p>
+    <p>Course: <%= link_to @course.name, supervisor_course_path(@course) %></p>
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/assigned_supervisors.html.erb]: i18n above

File: app/views/user_mailer/removed_supervisors.html.erb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
+    <h1>Welcome to <%= I18n.t :system_name %></h1>    
+    <p>
+      You have been removed from a course.
+    </p>
+    <p>Course: <%= link_to @course.name, supervisor_course_path(@course) %></p>
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/removed_supervisors.html.erb]: i18n above

File: config/environments/development.rb [hunk 7]
```diff
@@ -38,4 +38,17 @@
   config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
```
Comment (MuhammadTamzid) [inline config/environments/development.rb]: Convention and Double quote
{host: "localhost", port: 3000}

File: config/environments/development.rb [hunk 8]
```diff
@@ -38,4 +38,17 @@
...
+        port:                  587,
+        domain:                "gmail.com",
+        user_name:             "noreply.esv",
+        password:              "abrakada",
+        authentication:        :plain,
+        enable_starttls_auto:  true
+  }
+  config.action_mailer.perform_deliveries = true
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.default_options = {from: 'noreply.esv@gmail.com'}
```
Comment (MuhammadTamzid) [inline config/environments/development.rb]: Double quote

File: app/views/user_mailer/assigned_supervisors.html.erb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/assigned_supervisors.html.erb]: i18n

File: app/views/user_mailer/assigned_supervisors.html.erb [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    <h1>Welcome to <%= I18n.t :system_name %></h1>
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/assigned_supervisors.html.erb]: i18n

File: app/views/user_mailer/assigned_supervisors.html.erb [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
+    <h1>Welcome to <%= I18n.t :system_name %></h1>    
+    <p>
+      You have been assigned to a course.
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/assigned_supervisors.html.erb]: i18n

File: app/views/user_mailer/assigned_supervisors.html.erb (hunk 5 above)
Comment (MuhammadTamzid) [inline app/views/user_mailer/assigned_supervisors.html.erb]: i18n

File: app/views/user_mailer/removed_supervisors.html.erb [hunk 12]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/removed_supervisors.html.erb]: i18n

File: app/views/user_mailer/removed_supervisors.html.erb [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+    <h1>Welcome to <%= I18n.t :system_name %></h1>
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/removed_supervisors.html.erb]: i18n

File: app/views/user_mailer/removed_supervisors.html.erb [hunk 14]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
+    <h1>Welcome to <%= I18n.t :system_name %></h1>    
+    <p>
+      You have been removed from a course.
```
Comment (MuhammadTamzid) [inline app/views/user_mailer/removed_supervisors.html.erb]: i18n

File: app/views/user_mailer/removed_supervisors.html.erb (hunk 6 above)
Comment (MuhammadTamzid) [inline app/views/user_mailer/removed_supervisors.html.erb]: i18n

File: Gemfile [hunk 15]
```diff
@@ -31,6 +31,7 @@ gem "will_paginate", "~> 3.0.6"
 gem "will_paginate-bootstrap"
 gem "bootstrap-datepicker-rails"
 gem "cancancan"
+gem 'sidekiq'
```
Comment (dieunb) [inline Gemfile]: double quote

**Pull request merged**